### ISSUE TITLE
ai_trader.py

### ISSUE THREAD

**Issue by ahmed1ngm1** [microsoft/vscode-copilot-release#11023]
- Copilot Chat Extension Version:
- VS Code Version:
- OS Version:
- Feature (e.g. agent/edit/ask mode):
- Selected model (e.g. GPT 4.1, Claude 3.7 Sonnet):
- Logs:

Steps to Reproduce:

1.import pandas as pd
import numpy as np

SIGNAL_PATH = r"C:\Users\Qebaa\AppData\Roaming\MetaQuotes\Terminal\C419E35F6EFCAC731FACC09E5EB1D6BB\MQL4\Files\signals.txt"

def calculate_indicators(df):
    df['MA50'] = df['Close'].rolling(window=50).mean()
    df['MA200'] = df['Close'].rolling(window=200).mean()
    delta = df['Close'].diff()
    gain = np.where(delta > 0, delta, 0)
    loss = np.where(delta < 0, -delta, 0)
    avg_gain = pd.Series(gain).rolling(window=14).mean()
    avg_loss = pd.Series(loss).rolling(window=14).mean()
    rs = avg_gain / avg_loss
    df['RSI'] = 100 - (100 / (1 + rs))
    return df

def analyze_symbol(df):
    df = calculate_indicators(df)
    last = df.iloc[-1]
    if last['Close'] > last['MA50'] and last['RSI'] < 30:
        return 'BUY'
    elif last['Close'] < last['MA50'] and last['RSI'] > 70:
        return 'SELL'
    return None

def main():
    data = pd.read_csv('market_data.csv')
    signals = []

    for symbol in data['Symbol'].unique():
        df_symbol = data[data['Symbol'] == symbol].sort_values(['Date','Time'])
        decision = analyze_symbol(df_symbol)
        if decision:
            last_close = df_symbol.iloc[-1]['Close']
            signals.append(f"{symbol},{decision},{last_close}")

    if signals:
        with open(SIGNAL_PATH, 'w') as f:
            for signal in signals:
                f.write(signal + '\n')
        print(f"✅ تم إرسال {len(signals)} توصيات لـ MT4")
    else:
        print("❌ لا توجد توصيات الآن.")

if __name__ == '__main__':
    main()

2.
